src/core/positioning/egenioussreceiver.cpp

```diff
@@ -9,37 +9,34 @@ EgenioussReceiver::EgenioussReceiver( QObject *parent )
 {
   connect( mTcpSocket, &QTcpSocket::readyRead, this, &EgenioussReceiver::onReadyRead );
   connect( mTcpSocket, &QTcpSocket::errorOccurred, this, &EgenioussReceiver::handleError );
-  connect( mTcpSocket, &QTcpSocket::connected, this, &EgenioussReceiver::connected );
-  connect( mTcpSocket, &QTcpSocket::disconnected, this, &EgenioussReceiver::disconnected );
+  connect( mTcpSocket, &QTcpSocket::stateChanged, this, &EgenioussReceiver::setSocketState );
```

Comment:
```suggestion
  connect( mTcpSocket, &QTcpSocket::stateChanged, this, &AbstractGnssReceiver::stateChanged );
```

Here you can connect the signal of the socket directly with the signal of the abstract class, then you can delete the method `setSocketState`. You just need in the Abstract class to fire socketStateStringChanged